pkg/services/govmomi/service.go

```diff
@@ -71,15 +72,17 @@ func (vms *VMService) ReconcileVM(ctx *context.VMContext) (vm infrav1.VirtualMac
 	// Before going further, we need the VM's managed object reference.
 	vmRef, err := findVM(ctx)
 	if err != nil {
-		if !isNotFound(err) {
+		if !errors.As(err,find.NotFoundError{}) {
```

Comment:
We create a `errNotFound` when the err from the govomi lookup [comes back nil](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/master/pkg/services/govmomi/util.go#L61-L67) but the object reference is not set. With the original implementation, that would fall through this condition. With the new changes, it looks like it would return early as errNotFound won't have a `find.NotFoundError` in its chain.